### DISCUSSION TITLE
[Foundry] How to generate Standard JSON with foundry?

### DISCUSSION THREAD

**Discussion by wwwMalcolm** [zkSync-Community-Hub/zksync-developers#790] [Dev Tooling] [answered]
### Environment

Testnet

### Select the Dev Tool you are using

Foundry

### Provide the version of the tool (if applicable)

forge 0.0.2 (420660c 2024-11-21T00:27:54.875617523Z)

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

I want to be able to generate the Standard JSON so that I can test a compilation using binaries (.exe)'s instead.

**Comment by alhenaai** [comment]
To generate Standard JSON using Foundry for ZKsync, you can utilize the `forge build` command with specific options tailored for ZKsync environments. Here’s how you can do it:

1. **Set Up Your Foundry Environment**: Ensure that your Foundry environment is correctly set up for ZKsync. If you haven't set it up yet, you can find detailed instructions on the [Foundry Installation Guide](https://docs.zksync.io/build/tooling/foundry/installation).

2. **Generate Standard JSON**:
   - Use the `forge build` command with the `--zksync` option to compile your smart contracts into ZKsync VM bytecode.
   - The compiled files, including the Standard JSON, are stored in a structured directory at `<PROJECT-ROOT>/zkout/`.

Here is the command you might use:
```bash
forge build --zksync
```

This command compiles your smart contracts for the ZKsync environment and outputs the necessary JSON files for further testing or deployment.

For more detailed information on using Foundry with ZKsync, including testing and deploying smart contracts, you can refer to the [Getting Started Guide](https://docs.zksync.io/build/tooling/foundry/getting-started).

If you need further assistance or more in-depth discussion, consider visiting the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions).

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by wwwMalcolm** [reply]
The bot's command helps me to generate the output.json produced after using the Standard-Json method. I want to know what is the input.json USED to generate the output instead.

**Reply by alhenaai** [reply]
To generate the Standard JSON input for Foundry when working with ZKsync, you typically don't directly manipulate or create a `input.json` file as you might with other Solidity development environments like Hardhat or Truffle. Instead, Foundry handles much of the configuration and setup internally.

However, if you need to customize the input JSON for specific configurations or to integrate with other tools that require a Standard JSON input format, you would typically need to adjust the settings within your Foundry project's configuration files or through command-line options that influence how the compilation is handled.

Here are steps you can consider:

1. **Review Foundry Configuration**: Check your `foundry.toml` or similar configuration files to understand the current settings that might influence the generation of input JSON.

2. **Customize Build Options**: Use the `forge build` command with additional options to specify or override default settings. For example, you can use flags to set optimization levels or to include/exclude certain contracts.

3. **Extract Input JSON**:
   - While Foundry does not explicitly generate a visible `input.json` file by default, you can simulate or extract the equivalent JSON by understanding the compilation flags and settings used in your build commands.
   - If necessary, consider modifying the Foundry source or using a script to log the effective input JSON based on the runtime configuration and compilation context.

For more detailed control over the JSON input or if you're integrating with tools that require explicit JSON files, you might need to manually create a JSON file that mimics the settings and configurations Foundry uses, based on the documentation and the effective settings in your environment.

For further guidance on managing and customizing the compilation process in Foundry tailored to ZKsync, you can refer to the detailed documentation on [Foundry's compilation options](https://docs.zksync.io/build/tooling/foundry/getting-started#compilation-with-forge-build-zksync) or engage with the community for specific use cases at the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions).

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by kaymomin** [comment, marked as answer]
Hi @wwwMalcolm -- to generate the standard JSON, try using: 

`forge create --show-standard-json-input <CONTRACT>`

Note, the [standard JSON](https://github.com/matter-labs/foundry-zksync/blob/main/crates/forge/bin/cmd/create.rs#L89) can only be generated if the contract has been deployed.

**Reply by wwwMalcolm** [reply]
`forge create --show-standard-json-input Counter`
Hello @kaymomin I tried this command above but got the below error:

```
error: the following required arguments were not provided:
  --verify

Usage: forge create --verify --show-standard-json-input <CONTRACT>

For more information, try '--help'.
```

Even if I did the following `forge create --verify --show-standard-json-input Counter` I would get the below error.

```
Error: error sending request for url (http://localhost:8545/)

Context:
- Error #0: error sending request for url (http://localhost:8545/)
- Error #1: client error (Connect)
- Error #2: tcp connect error: Connection refused (os error 111)
- Error #3: Connection refused (os error 111)
```
